### ISSUE TITLE
Feature request: specify failure job

### ISSUE THREAD

**Issue by rcarmich** [rq/rq#1441]
Currently I have a scheduled job that reviews the FailedJobRegistry at some interval to attempt to handle failures gracefully. However, it would be really helpful if we could specify a job to run if any job in my depends_on chain fails - like a "finally" clause in try/catch.

For example I have an entity stored in Redis that keeps a Status field updated for each job in a series of steps. It would be great to just be able to specify at the top level a job to run if any exceptions are thrown so that job could mark my entity's status as "Failed".

I realize you can do worker-level failure handlers, but this approach gives more freedom as it is defined per job.

**Comment by selwin** [comment]
Yes, I agree. Actually adding a success/failure callback is high on my priority list. It would be nice to be able to do something like `queue.enqueue(foo, on_success=bar, on_failure=baz)`.

I haven't had time to do this yet, but would gladly accept a PR for a feature like this!

**Comment by dfrank8** [comment]
This would be REALLY cool! Currently, we use a super sophisticated exception handler.

In `settings.py`:
```
RQ_EXCEPTION_HANDLERS = [
    'my.app.django_rq_custom_failed'
]
```

It does a few things that you mentioned, such as:
- enqueues retries under certain circumstances. we pass certain fields to the `meta` to tell the exception handler to do certain stuff:
```
        low_queue.enqueue(
            f=my_function,
            field=value,
            meta={
                "retry_on": [
                    ConnectionError,
                    HTTPError
                ],
                "retry_wait": 5,
                "exponential_retry": True
            }
```
- We also check if there are `depends_on` jobs, and will enqueue the next in the chain if the job fails, since any failed job breaks the whole `depends_on` chain.
- Sends alerts/notifications to some Slack channels if the failure is really bad

Obviously, there's no built-in solution for `on_success` like Selwin said, but for failures you can lean heavily on this exception handler.

**Comment by 56907dzq** [comment]
how to add args to on_success?

**Comment by 56907dzq** [comment]
I noticed def create(cls, func, args=None, kwargs=None, connection=None,
               result_ttl=None, ttl=None, status=None, description=None,
               depends_on=None, timeout=None, id=None, origin=None, meta=None,
               failure_ttl=None, serializer=None, *, on_success=None, on_failure=None):
*,add param directly? or something,
looks it work
